Marlin/boards.h

```diff
@@ -76,6 +76,7 @@
 #define BOARD_BQ_ZUM_MEGA_3D    503   // bq ZUM Mega 3D
 #define BOARD_MAKEBOARD_MINI    431   // MakeBoard Mini v2.1.2 is a control board sold by MicroMake
 #define BOARD_TRIGORILLA        343   // TriGorilla Anycubic version 1.3 based on RAMPS EFB
+#define BOARD_RAMPS_ENDER_4     49    // Creality: Ender-4, CR-8
```

Comment:
This conflicts with `BOARD_SAINSMART_2IN1`. We'll have to pick a different ID. But, isn't the Ender 4 board just the same as `BOARD_MELZI_CREALITY`?

Reply:
Oops. Not sure how I didn't spot that.

Is there any particular convention for choosing these IDs?

Reply:
> But, isn't the Ender 4 board just the same as `BOARD_MELZI_CREALITY`?

Nope. It's mostly RAMPS-equivalent, but there is one oddity that may actually be related to the issue you spotted. I'm diagnosing it now.